src/MuTalk-Model/FixedRandomOperatorMutantSelection.class.st

```diff
@@ -0,0 +1,32 @@
+Class {
+	#name : #FixedRandomOperatorMutantSelection,
+	#superclass : #FixedRandomMutantSelection,
+	#category : #'MuTalk-Model-Mutations generation strategies'
+}
+
+{ #category : #accessing }
+FixedRandomOperatorMutantSelection >> selectMutantsFrom: aCollection [
+
+	| size index newColl random dict operators operator |
+	newColl := aCollection copyEmpty.
+	random := Random new.
+
+	numberOfMutants = 0 ifTrue: [ ^ newColl ].
+	aCollection size <= numberOfMutants ifTrue: [ ^ aCollection ].
+
+	dict := aCollection groupedBy: [ :e | e operator ].
+	operators := dict keys.
+	1 to: numberOfMutants do: [ :i |
+		operator := operators at:
+			            (random nextIntegerBetween: 1 and: operators size).
+		[ (dict at: operator) allSatisfy: [ :each | newColl includes: each ] ]
+			whileTrue: [
+				operator := operators at:
+					            (random nextIntegerBetween: 1 and: operators size) ].
+		size := (dict at: operator) size.
```

Comment:
This looks like the same code as above. How can we enhance it a bit and avoid the repeated code?